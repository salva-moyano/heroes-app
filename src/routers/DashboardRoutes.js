import {Navbar} from "../components/ui/NavBar";
import {Routes, Route} from "react-router-dom";
import {MarvelScreen} from "../components/marvel/MarvelScreen";
import {DCScreen} from "../components/dc/DCScreen";
import {SearchScreen} from "../components/search/SearchScreen";
import {HeroScreen} from "../components/hero/HeroScreen";

export const DashboardRoutes = () => {
    return (
        <>
          <Navbar/>
            <div className="container">
                <Routes>
                    <Route path="marvel" element={<MarvelScreen/>}/>
                    <Route path="dc" element={<DCScreen/>}/>
                    <Route path="search" element={<SearchScreen/>}/>
                    <Route path="hero/:heroId" element={<HeroScreen/>}/>

                    <Route path="/" element={<MarvelScreen/>}/>
                </Routes>
            </div>
        </>
    )
}