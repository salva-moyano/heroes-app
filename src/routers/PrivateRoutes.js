import {AuthContext} from "../auth/authContext";
import { useContext } from "react";
import {Navigate, useLocation} from "react-router-dom";

export const PrivateRoutes = ({ children }) => {

    const { user } = useContext(AuthContext);

    const {pathname, search} = useLocation();

    localStorage.setItem('lastPath', `${pathname}${search}`);

    return user.logged ? children : <Navigate to="/login"/>
}