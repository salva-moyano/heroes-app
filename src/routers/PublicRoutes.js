import {useContext} from "react";
import {AuthContext} from "../auth/authContext";
import {Navigate} from "react-router-dom";

export const PublicRoutes = ({children}) => {
    const { user } = useContext(AuthContext);
    return user.logged ? <Navigate to="/marvel"/> : children;
}