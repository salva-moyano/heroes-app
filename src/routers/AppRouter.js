import {BrowserRouter, Route, Routes} from "react-router-dom";
import {LoginScreen} from "../components/login/LoginScreen";
import {DashboardRoutes} from "./DashboardRoutes";
import { PrivateRoutes } from "./PrivateRoutes";
import {PublicRoutes} from "./PublicRoutes";

export const AppRouter = () => {
    return(
        <BrowserRouter>

            <Routes>

                {/* <Route path="/login" element={<LoginScreen/>}/> */}

                <Route path="/login" element={
                    <PublicRoutes>
                        <LoginScreen/>
                    </PublicRoutes>
                }/>


                <Route path="/*" element={
                    <PrivateRoutes>
                        <DashboardRoutes/>
                    </PrivateRoutes>
                } />
                {/*  <Route path="/*" element={<DashboardRoutes/>}/>  */}
            </Routes>
        </BrowserRouter>
    )
}