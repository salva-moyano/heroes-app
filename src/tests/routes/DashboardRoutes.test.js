import { mount } from "enzyme";
import {DashboardRoutes} from "../../routers/DashboardRoutes";
import {AuthContext} from "../../auth/authContext";
import { MemoryRouter } from "react-router-dom";


describe('Testing <DashboardRoutes', () => {

    const contextValue = { user: { name: 'salva', logged: true }};

    test('should show successfully', () => {
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter initialEntries={['/']}>
                    <DashboardRoutes/>
                </MemoryRouter>
            </AuthContext.Provider>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('.text-info').text().trim()).toBe('salva');
        expect(wrapper.find('h1').text().trim()).toBe('Marvel Comics');
    });

    test('should show DC Screen successfully', () => {
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter initialEntries={['/dc']}>
                    <DashboardRoutes/>
                </MemoryRouter>
            </AuthContext.Provider>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('.text-info').text().trim()).toBe('salva');
        expect(wrapper.find('h1').text().trim()).toBe('DC Comics')
    });

    test('should show Search Screen successfully', () => {
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter initialEntries={['/search']}>
                    <DashboardRoutes/>
                </MemoryRouter>
            </AuthContext.Provider>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('.text-info').text().trim()).toBe('salva');

        expect(wrapper.find('h4').first().text().trim()).toBe('Buscar');
    });

    test('should show Hero Screen successfully', () => {

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter initialEntries={['/hero/dc-batman']}>
                    <DashboardRoutes/>
                </MemoryRouter>
            </AuthContext.Provider>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('.text-info').text().trim()).toBe('salva');
        expect(wrapper.find('h3').text().trim()).toBe('Batman');
    });

});