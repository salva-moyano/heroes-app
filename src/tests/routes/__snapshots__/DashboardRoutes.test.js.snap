// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <DashboardRoutes should show DC Screen successfully 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Heroes
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link false"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link false"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          salva
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <div
        className="row animate__animated animate__fadeIn"
      >
        <h1>
          DC Comics
        </h1>
        <hr />
        <div
          className="col-3"
          key="dc-batman"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Batman"
                  className="card-img"
                  src="../assets/heroes/dc-batman.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Batman
                  </h5>
                  <p
                    className="card-text"
                  >
                    Bruce Wayne
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      Detective Comics #27
                    </small>
                  </p>
                  <a
                    href="/hero/dc-batman"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="dc-superman"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Superman"
                  className="card-img"
                  src="../assets/heroes/dc-superman.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Superman
                  </h5>
                  <p
                    className="card-text"
                  >
                    Kal-El
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      Action Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/dc-superman"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="dc-flash"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Flash"
                  className="card-img"
                  src="../assets/heroes/dc-flash.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Flash
                  </h5>
                  <p
                    className="card-text"
                  >
                    Jay Garrick
                  </p>
                  <p
                    className="text-muted"
                  >
                    Jay Garrick, Barry Allen
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      Flash Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/dc-flash"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="dc-green"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Green Lantern"
                  className="card-img"
                  src="../assets/heroes/dc-green.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Green Lantern
                  </h5>
                  <p
                    className="card-text"
                  >
                    Alan Scott
                  </p>
                  <p
                    className="text-muted"
                  >
                    Alan Scott, Hal Jordan
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      All-American Comics #16
                    </small>
                  </p>
                  <a
                    href="/hero/dc-green"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="dc-arrow"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Green Arrow"
                  className="card-img"
                  src="../assets/heroes/dc-arrow.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Green Arrow
                  </h5>
                  <p
                    className="card-text"
                  >
                    Oliver Queen
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      More Fun Comics #73
                    </small>
                  </p>
                  <a
                    href="/hero/dc-arrow"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="dc-wonder"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Wonder Woman"
                  className="card-img"
                  src="../assets/heroes/dc-wonder.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Wonder Woman
                  </h5>
                  <p
                    className="card-text"
                  >
                    Princess Diana
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      All Star Comics #8
                    </small>
                  </p>
                  <a
                    href="/hero/dc-wonder"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="dc-martian"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Martian Manhunter"
                  className="card-img"
                  src="../assets/heroes/dc-martian.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Martian Manhunter
                  </h5>
                  <p
                    className="card-text"
                  >
                    J'onn J'onzz
                  </p>
                  <p
                    className="text-muted"
                  >
                    Martian Manhunter
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      Detective Comics #225
                    </small>
                  </p>
                  <a
                    href="/hero/dc-martian"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="dc-robin"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Robin/Nightwing"
                  className="card-img"
                  src="../assets/heroes/dc-robin.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Robin/Nightwing
                  </h5>
                  <p
                    className="card-text"
                  >
                    Dick Grayson
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      Detective Comics #38
                    </small>
                  </p>
                  <a
                    href="/hero/dc-robin"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="dc-blue"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Blue Beetle"
                  className="card-img"
                  src="../assets/heroes/dc-blue.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Blue Beetle
                  </h5>
                  <p
                    className="card-text"
                  >
                    Dan Garret
                  </p>
                  <p
                    className="text-muted"
                  >
                    Dan Garret, Ted Kord, Jaime Reyes
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      Mystery Men Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/dc-blue"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="dc-black"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Black Canary"
                  className="card-img"
                  src="../assets/heroes/dc-black.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Black Canary
                  </h5>
                  <p
                    className="card-text"
                  >
                    Dinah Drake
                  </p>
                  <p
                    className="text-muted"
                  >
                    Dinah Drake, Dinah Lance
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      Flash Comics #86
                    </small>
                  </p>
                  <a
                    href="/hero/dc-black"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Testing <DashboardRoutes should show Hero Screen successfully 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Heroes
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link false"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link false"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link false"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          salva
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div
      className="row mt-5"
    >
      <div
        className="col-4"
      >
        <img
          alt="Batman"
          className="img-thumbnail animate__animated animate__fadeInLeft"
          src="../assets/heroes/dc-batman.jpg"
        />
      </div>
      <div
        className="col-8 animate__animated animate__fadeIn"
      >
        <h3>
          Batman
        </h3>
        <ul
          className="list-group list-group-flush"
        >
          <li
            className="list-group-item"
          >
            <b>
               Alter ego: 
            </b>
             
            Bruce Wayne
          </li>
          <li
            className="list-group-item"
          >
            <b>
               Publisher: 
            </b>
             
            DC Comics
          </li>
          <li
            className="list-group-item"
          >
            <b>
               First Appearance: 
            </b>
             
            Detective Comics #27
          </li>
        </ul>
        <h5
          className="mt-3"
        >
          Characters
        </h5>
        <p>
          Bruce Wayne
        </p>
        <button
          className="btn btn-outline-info"
          onClick={[Function]}
        >
          Regresar
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`Testing <DashboardRoutes should show Search Screen successfully 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Heroes
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link false"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link false"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          salva
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-5"
      >
        <h4>
          Buscar
        </h4>
        <hr />
        <form
          onSubmit={[Function]}
        >
          <input
            autoComplete="off"
            className="form-control"
            name="searchText"
            onChange={[Function]}
            placeholder="Buscar un héroe"
            type="text"
            value=""
          />
          <div
            className="d-grid gap-2"
          >
            <button
              className="btn btn-outline-primary mt-3"
              type="submit"
            >
              Buscar...
            </button>
          </div>
        </form>
      </div>
      <div
        className="col-7"
      >
        <h4>
          Resultados
        </h4>
        <div
          className="alert alert-info animate__animated animate__fadeIn"
        >
          Buscar un héroe
        </div>
        <hr />
      </div>
    </div>
  </div>,
]
`;

exports[`Testing <DashboardRoutes should show successfully 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Heroes
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link false"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link false"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link false"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          salva
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <div
        className="row animate__animated animate__fadeIn"
      >
        <h1>
          Marvel Comics
        </h1>
        <hr />
        <div
          className="col-3"
          key="marvel-spider"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Spider Man"
                  className="card-img"
                  src="../assets/heroes/marvel-spider.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Spider Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Peter Parker
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      Amazing Fantasy #15
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-spider"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="marvel-captain"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Captain America"
                  className="card-img"
                  src="../assets/heroes/marvel-captain.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Captain America
                  </h5>
                  <p
                    className="card-text"
                  >
                    Steve Rogers
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      Captain America Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-captain"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="marvel-iron"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Iron Man"
                  className="card-img"
                  src="../assets/heroes/marvel-iron.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Iron Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Tony Stark
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      Tales of Suspense #39
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-iron"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="marvel-thor"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Thor"
                  className="card-img"
                  src="../assets/heroes/marvel-thor.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Thor
                  </h5>
                  <p
                    className="card-text"
                  >
                    Thor Odinson
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      Journey into Myster #83
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-thor"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="marvel-hulk"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Hulk"
                  className="card-img"
                  src="../assets/heroes/marvel-hulk.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hulk
                  </h5>
                  <p
                    className="card-text"
                  >
                    Bruce Banner
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      The Incredible Hulk #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hulk"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="marvel-wolverine"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Wolverine"
                  className="card-img"
                  src="../assets/heroes/marvel-wolverine.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Wolverine
                  </h5>
                  <p
                    className="card-text"
                  >
                    James Howlett
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      The Incredible Hulk #180
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-wolverine"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="marvel-daredevil"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Daredevil"
                  className="card-img"
                  src="../assets/heroes/marvel-daredevil.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Daredevil
                  </h5>
                  <p
                    className="card-text"
                  >
                    Matthew Michael Murdock
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      Daredevil #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-daredevil"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="marvel-hawkeye"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Hawkeye"
                  className="card-img"
                  src="../assets/heroes/marvel-hawkeye.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hawkeye
                  </h5>
                  <p
                    className="card-text"
                  >
                    Clinton Francis Barton
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      Tales of Suspense #57
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hawkeye"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="marvel-cyclops"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Cyclops"
                  className="card-img"
                  src="../assets/heroes/marvel-cyclops.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Cyclops
                  </h5>
                  <p
                    className="card-text"
                  >
                    Scott Summers
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      X-Men #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-cyclops"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-3"
          key="marvel-silver"
        >
          <div
            className="card animate__animated animate__fadeIn"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-6"
              >
                <img
                  alt="Silver Surfer"
                  className="card-img"
                  src="../assets/heroes/marvel-silver.jpg"
                />
              </div>
              <div
                className="col-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Silver Surfer
                  </h5>
                  <p
                    className="card-text"
                  >
                    Norrin Radd
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                       
                      The Fantastic Four #48
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-silver"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
